src/cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs

```diff
@@ -248,6 +249,27 @@ public static void CopyTo(this IBuffer source, uint sourceIndex, IBuffer destina
 
 #endregion (IBuffer).CopyTo extensions for copying to an (IBuffer)
 
+        public static bool TryGetDataUnsafe(this IBuffer buffer, out IntPtr dataPtr)
+        {
+            if (buffer is IWinRTObject winrtObj && winrtObj.HasUnwrappableNativeObject &&
+                winrtObj.NativeObject.TryAs<IUnknownVftbl>(global::ABI.Windows.Storage.Streams.IBufferByteAccessMethods.IID, out var objRef) >= 0)

```

Comment:
We can make this more efficient by just using the `TryAs` overload returning a raw pointer and doing the marshalling manually, no? That way we can avoid allocating a new `ObjectReference`, since it's not really needed anyway here. Basically we can just do manual RAII on that returned COM object.

Reply:
Not anymore as I'm using ComWrapperSupport now.

Reply:
But that still has the same problem. Shouldn't we keep the faster path on modern .NET?

Reply:
That would certainly be ideal. @manodasanW any suggestion for how to do that?

Reply:
So `TryUnwrapObject` should be optimized already for .NET 5+.  But you can avoid the object construction on the QI afterwards using the other `TryAs` method that gives you the ptr as an out and then update the methods static class to take a `IntPtr` instead.  You would need to call Marshal.Release afterwards.  There is also the `AsValue` method which takes an `IID` that achieves this, but it throws when QI fails so you probably don't want that here.